Sources/Helpers/Factories/ViewCellItemsFactory.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  Copyright © 2020 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+open class ViewCellItemsFactory<Object: GenericDiffItem, View: UIView> {
+
+    public typealias Cell = CollectionViewViewCell<View>
+    public typealias CellItem = CollectionViewViewCellItem<Object, View>
+    typealias AssociatedFactory = AssociatedCellItemFactory<Object, Cell>
+
+    /// Set this handler to retrieve a specific set of cell items for the associated object
+    ///
+    /// - Parameters:
+    ///    - Object: the object associated with a cell item
+    public var initializationHandler: ((Object) -> [CollectionViewCellItem?])?
+
+    /// Set this handler to configure the cell item
+    ///
+    /// - Parameters:
+    ///    - CellItem: generated universal cell item or defined in `initializationHandler`  for this object
+    public var cellItemConfigurationHandler: ((CellItem) -> Void)?
+
+    /// Set this handler to configure the size of cell
+    ///
+    /// - Parameters:
+    ///    - Object: the object associated with a cell item
+    ///    - UICollectionView: collection view where cell should be placed
+    ///    - CollectionViewSectionItem: a section item in the section of which the cell should be placed
+    public var sizeConfigurationHandler: ((Object, UICollectionView, CollectionViewSectionItem) -> CGSize)? {
+        didSet {
+            factory.sizeConfigurationHandler = sizeConfigurationHandler
+        }
+    }
+
+    /// Set this handler to provide size types for cellItem
+    ///
+    /// - Parameters:
+    ///    - Object: the object associated with a cell item
+    public var sizeTypesConfigurationHandler: ((Object) -> SizeTypes)?
+
+    /// Set this handler to provide specific an instance of `View`
+    ///
+    /// - Parameters:
+    ///    - CellItem: generated universal cell item or defined in `initializationHandler`  for this object
+    public var viewInitializer: ((CellItem) -> View)? = { _ in
+        .init()
+    }
```

Comment:
And here would be better to provide an object as a second parameter

Reply:
same here - object stored in cellItem